include/deal.II/base/tensor.h

```diff
@@ -1357,6 +1357,45 @@ Tensor<rank_, dim, Number>::component_to_unrolled_index(
 }
 
 
+
+namespace internal
+{
+  // unrolled_to_component_indices is instantiated from DataOut for dim==0
+  // and rank=2. Make sure we don't have compiler warnings.
+
+  template <int dim>
+  inline unsigned int
+  mod(const unsigned int &remainder)
+  {
+    return remainder % dim;
```

Comment:
`remainder` is an odd choice for the variable name here because the variable is the *operand*, whereas the remainder is the *result* of the operation. Just call it `x` or something. Also, no need for the reference.